frontend/pages/ManageControlsPage/Scripts/components/EditScriptModal/EditScriptModal.tsx

```diff
@@ -0,0 +1,143 @@
+import { DEFAULT_USE_QUERY_OPTIONS } from "utilities/constants";
+import { getErrorMessage } from "../ScriptUploader/helpers";
+
+const baseClass = "edit-script-modal";
+
+interface IEditScriptModal {
+  onExit: () => void;
+  scriptId: number;
+  scriptName: string;
+}
+
+const EditScriptModal = ({
+  scriptId,
+  scriptName,
+  onExit,
+}: IEditScriptModal) => {
+  const { renderFlash } = useContext(NotificationContext);
+
+  // Editable script content
+  const [scriptFormData, setScriptFormData] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [formError, setFormError] = useState<string | null>(null);
+
+  const {
+    error: isSelectedScriptContentError,
+    isLoading: isLoadingSelectedScriptContent,
+  } = useQuery<ScriptContent, Error>(
+    [scriptId],
+    () => scriptAPI.downloadScript(scriptId),
+    {
+      ...DEFAULT_USE_QUERY_OPTIONS,
+      onSuccess: (scriptContent) => {
+        setScriptFormData(scriptContent);
+      },
+    }
+  );
+
+  const validate = (scriptContent: string) => {
+    if (scriptContent.trim() === "") {
+      setFormError("Script cannot be empty");
+    } else {
+      setFormError(null);
+    }
+  };
+
+  const onChange = (value: string) => {
+    setScriptFormData(value);
+    validate(value);
+  };
+
+  const onSave = async () => {
+    if (isSubmitting) {
+      return;
+    }
+    try {
+      setIsSubmitting(true);
+      await scriptAPI.updateScript(scriptId, scriptFormData, scriptName);
+      renderFlash("success", "Successfully saved script.");
+      onExit();
+    } catch (e) {
+      renderFlash("error", getErrorMessage(e));
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  const onSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    onSave();
+  };
+
+  const renderContent = () => {
+    if (isLoadingSelectedScriptContent) {
+      return <Spinner />;
+    }
+
+    if (isSelectedScriptContentError) {
+      return <DataError description="Close this modal and try again." />;
+    }
+
+    return (
+      <>
+        <form onSubmit={onSubmit}>
+          <Editor
+            value={scriptFormData}
+            onChange={onChange}
+            isFormField
+            error={formError}
+          />
+          <div className="form-field__help-text">
+            To run this script on a host, go to the{" "}
+            <CustomLink text="Hosts" url={paths.MANAGE_HOSTS} /> page and select
+            a host.
+            <br />
+            To run the script across multiple hosts, add a policy automation on
+            the <CustomLink text="Policies" url={paths.MANAGE_POLICIES} /> page.
+          </div>
+        </form>
+        <ModalFooter
+          primaryButtons={
+            <>
+              <Button onClick={onExit} variant="inverse">
+                Cancel
+              </Button>
+              <Button onClick={onSave} variant="brand" isLoading={isSubmitting}>
```

Comment:
Let's also set `disabled` to `true` if there are any errors (`disabled={!!formError}` as currently
implemented, or generally `disabled={Object.keys(formErrors).length > 0}` with an error object implementation